packages/gestalt/dist/index.d.ts

```diff
@@ -1729,23 +1729,28 @@ interface TableSortableHeaderCellProps {
 
 interface TableRowProps {
   children: Node;
+  hoverStyle?: 'gray' | 'none' | undefined;
+  selected?: 'selected' | 'unselected' | undefined;
 }
 
 interface TableRowExpandableProps {
   accessibilityCollapseLabel: string;
   accessibilityExpandLabel: string;
   children: Node;
   expandedContents: Node;
-  id: string;
   expanded?: string | undefined;
-  hoverStyle?: 'none' | 'gray' | undefined;
+  hoverStyle?: 'gray' | 'none' | undefined;
```

Comment:
Interesting…is this correct? It's different from the Flow type. Flow is saying "this prop is optional, but if it's defined, it will be this enum of values". TS is saying "this prop is optional, but if it's defined, it will be this enum of values or `undefined`". The TS type seems to give additional flexibility we're not intending.

Reply:
Good catch. I'll leave because I kept that pattern of optional props having undefined value, but I realized later that they are unnecessary. Technically optional, if not passed, is undefined and defaults props override undefined.

I'll clean up this issue in a follow up but for all types.